pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/CopyConfigurationDao.java

```diff
@@ -0,0 +1,184 @@
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+import org.jdbi.v3.sqlobject.CreateSqlObject;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterConstructorMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.UseRowReducer;
+import org.jdbi.v3.stringtemplate4.UseStringTemplateSqlLocator;
+
+public interface CopyConfigurationDao extends SqlObject {
+
+    @CreateSqlObject
+    CopyConfigurationSql getCopyConfigurationSql();
+
+    default CopyConfiguration createCopyConfig(CopyConfiguration config) {
+        long configId = getCopyConfigurationSql().insertCopyConfig(config.getStudyId());
+        int order = 1;
+        for (var pair : config.getPairs()) {
+            pair.setOrder(order);
+            createCopyPair(config.getStudyId(), configId, pair);
+            order += 1;
+        }
+        return findCopyConfigById(configId).orElseThrow(() ->
+                new DaoException("Could not find newly created copy configuration with id " + configId));
+    }
+
+    private long createCopyPair(long studyId, long configId, CopyConfigurationPair pair) {
+        if (pair.getSource().getType() != CopyLocationType.ANSWER) {
+            throw new DaoException("Currently only answer source locations are supported");
+        }
+
+        var compositeToComposite = false;
+        var jdbiQuestion = getHandle().attach(JdbiQuestion.class);
+
+        var source = (CopyAnswerLocation) pair.getSource();
+        QuestionDto sourceQuestionDto = jdbiQuestion
+                .getLatestQuestionDtoByQuestionStableIdAndUmbrellaStudyId(source.getQuestionStableId(), studyId)
+                .orElseThrow(() -> new DaoException(String.format(
+                        "Could not find source question with stable id %s in study %d",
+                        source.getQuestionStableId(), studyId)));
+        if (sourceQuestionDto.getType() == QuestionType.COMPOSITE && pair.getTarget().getType() != CopyLocationType.ANSWER) {
+            throw new DaoException("Currently only support copying from source composite to target composite questions");
+        }
+
+        if (pair.getTarget().getType() == CopyLocationType.ANSWER) {
+            var target = (CopyAnswerLocation) pair.getTarget();
+            QuestionDto targetQuestionDto = jdbiQuestion
+                    .getLatestQuestionDtoByQuestionStableIdAndUmbrellaStudyId(target.getQuestionStableId(), studyId)
+                    .orElseThrow(() -> new DaoException(String.format(
+                            "Could not find target question with stable id %s in study %d",
+                            target.getQuestionStableId(), studyId)));
+            if (sourceQuestionDto.getType() != targetQuestionDto.getType()) {
+                throw new DaoException("Currently only support copying answers between questions of the same type");
```

Comment:
Since implementing every combination might be a lot of work, I have added some restrictions on what pairings we can create. And looking at the copying primitives we have added in #40, it looks like we only allow copying answers between same question types.